dropwizard/service/src/main/java/org/apache/polaris/service/dropwizard/auth/PolarisPrincipalRoleSecurityContextProvider.java

```diff
@@ -53,6 +54,10 @@ public void filter(ContainerRequestContext requestContext) throws IOException {
   public SecurityContext createSecurityContext(
       SecurityContext ctx, AuthenticatedPolarisPrincipal principal) {
     Set<String> validRoleNames = activeRolesProvider.getActiveRoles(principal);
+    if (validRoleNames.isEmpty()) {
```

Comment:
This is making the tests fail, and I think we do have a problem.  `DefaultActiveRolesProvider` expects the principal to have the role `PRINCIPAL_ROLE_ALL` when all roles are activated:

```java
boolean allRoles = tokenRoles.contains(BasePolarisAuthenticator.PRINCIPAL_ROLE_ALL);
```

But `BasePolarisAuhtenticator` creates a principal with an empty roles set in that case. You need to change `BasePolarisAuthenticator.getPrincipal()`:

```java
  if (tokenInfo.getScope() != null) {
      if (tokenInfo.getScope().equals(PRINCIPAL_ROLE_ALL)) {
        activatedPrincipalRoles.add(PRINCIPAL_ROLE_ALL); // this is missing
      } else {
        activatedPrincipalRoles.addAll(...);
      }
    }
```